app/javascript/packs/application.js

```diff
@@ -2,11 +2,13 @@
 // present in this directory. You're encouraged to place your actual application logic in
 // a relevant structure within app/javascript and only use these pack files to reference
 // that code so it'll be compiled.
-
 require("@rails/ujs").start()
 require("turbolinks").start()
 require("@rails/activestorage").start()
 require("channels")
+require("packs/custom")
+require("packs/posts.js")
+require("jquery")
```

Comment:
why?

Reply:
While adding popper.js and jquery with bootstrap I think so that may be thses both files are must need to require while writting require('bootstrap').But now I have understand that all three are independent.

Reply:
no, bootstrap depends on poper.js and jquery.js